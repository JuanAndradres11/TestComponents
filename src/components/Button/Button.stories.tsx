// src/components/ButtonComponent.stories.tsx
import React from 'react';
import { type Meta, type StoryObj } from '@storybook/react';
import ButtonComponent from './button';

const meta: Meta<typeof ButtonComponent> = {
  title: 'Components/Button',
  component: ButtonComponent,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof ButtonComponent>;

export const Primary: Story = {
  args: {
    label: 'Click Me',
    onClick: () => alert('Button clicked!'),
  },
};

export const Disabled: Story = {
  args: {
    label: 'Disabled',
    disabled: true,
  },
};
