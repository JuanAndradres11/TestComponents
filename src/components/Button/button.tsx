// src/components/Button/Button.tsx
import React from "react";
import MuiButton from "@mui/material/Button";

interface ButtonProps {
  label: string;
  onClick: () => void;
  variant?: "text" | "contained" | "outlined";
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({
  label,
  onClick,
  variant = "contained",
  disabled = false,
}) => {
  return (
    <MuiButton 
      variant={variant} 
      onClick={onClick} 
      disabled={disabled}
    >
      {label}
    </MuiButton>
  );
};

export default Button;
