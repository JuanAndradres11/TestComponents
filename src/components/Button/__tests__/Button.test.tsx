import { render, screen, fireEvent } from "@testing-library/react";
import Button from "../Button";

describe("Button", () => {
  it("renders with text", () => {
    render(<Button>Click Me</Button>);
    expect(screen.getByText("Click Me")).toBeInTheDocument();
  });

  it("triggers onClick when clicked", () => {
    const onClick = vi.fn();
    render(<Button onClick={onClick}>Press</Button>);
    fireEvent.click(screen.getByText("Press"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("applies disabled state", () => {
    const onClick = vi.fn();
    render(<Button disabled onClick={onClick}>Disabled</Button>);
    const btn = screen.getByText("Disabled");
    fireEvent.click(btn);
    expect(onClick).not.toHaveBeenCalled();
    expect(btn).toBeDisabled();
  });
});
