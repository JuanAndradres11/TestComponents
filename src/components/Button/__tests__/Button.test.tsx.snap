// Jest Snapshot v1, for Button.test.tsx
exports[`Button click triggers correctly 1`] = `
<button>Submit</button>
`;
